pyadjoint/optimization/rol_solver_checkpointed.py

```diff
@@ -0,0 +1,283 @@
+"""
+    Checkpointed optimisation with ROL2.0
+"""
+import os
+import shutil
+
+from ..tape import no_annotations
+from mpi4py import MPI
+from .rol_solver import (
+    ROLVector, ROLObjective, ROLSolver)
+from .optimization_solver import OptimizationSolver
+from firedrake.checkpointing import CheckpointFile
+from firedrake import utils
+
+vector_registry = []
+
+try:
+    import ROL
+
+    class __ROLCheckpointManager__(object):
```

Comment:
I don't really like having a "dunder" (i.e. `__xxx___`) class name, prefer just a private class `_ROLCheckpointManager`. Same goes throughout the file: single underscore prefix rather than dunders.